openapi3filter/issue884_test.go

```diff
@@ -0,0 +1,113 @@
+package openapi3filter
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/routers/gorillamux"
+)
+
+func TestIssue884(t *testing.T) {
+	loader := openapi3.NewLoader()
+	ctx := loader.Context
+	spec := `
+  openapi: 3.0.0
+  info:
+    version: 1.0.0
+    title: Sample API
+  components:
+    schemas:
+      TaskSortEnum:
+        enum:
+          - createdAt
+          - -createdAt
+          - updatedAt
+          - -updatedAt
+  paths:
+    /tasks:
+      get:
+        operationId: ListTask
+        parameters:
+          - in: query
+            name: withDefault
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/TaskSortEnum'
+                - default: -createdAt
+          - in: query
+            name: withoutDefault
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/TaskSortEnum'
+          - in: query
+            name: withManyDefaults
+            schema:
+              allOf:
```

Comment:
this definition should be schema validation error ?

Reply:
I'm unable to find in the spec à justification to "forbid multiple schema defaults". But if you find one then yes let's impl this validation pass. 

Otherwise let's continue to not return a validation error here.

Reply:
Yes, it is not required according to the specifications.